qt/scientific_interfaces/Direct/observerPattern.h

```diff
@@ -0,0 +1,63 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2014 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTIDQT_CUSTOMINTERFACES_OBSERVERPATTERN_H_
+#define MANTIDQT_CUSTOMINTERFACES_OBSERVERPATTERN_H_
+#include <functional>
+#include <set>
+#include <string>
+
+class observable;
```

Comment:
Can we follow the coding standards and use capital letters for first letters of class names?

Is there a reason that we're re-implementing the observer pattern and not just using signal/slots or the Poco Notification system?

Reply:
I wanted something that we can mock out later for testing. After talking to Gemma the observer pattern seemed the way to go.

Reply:
Ok. It might be worth moving this to common if we start using this more frequently.

I think it could do with some documentation though if possible?

Reply:
I will add something brief in developer docs, although I think we have some already on the general concept.

Reply:
I actually just meant doxygen in the file as class/method docs. I think that will be fine.